WordPress/src/main/java/org/wordpress/android/localcontentmigration/LocalMigrationContentResolver.kt

```diff
@@ -0,0 +1,70 @@
+package org.wordpress.android.localcontentmigration
+
+import android.content.ContentResolver
+import android.database.Cursor
+import android.net.Uri
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.PostActionBuilder
+import org.wordpress.android.localcontentmigration.LocalContentEntityData.EligibilityStatusData
+import org.wordpress.android.localcontentmigration.LocalContentEntity.EligibilityStatus
+import org.wordpress.android.localcontentmigration.LocalContentEntity.Post
+import org.wordpress.android.localcontentmigration.LocalContentEntity.Site
+import org.wordpress.android.localcontentmigration.LocalContentEntityData.PostData
+import org.wordpress.android.localcontentmigration.LocalContentEntityData.PostsData
+import org.wordpress.android.localcontentmigration.LocalContentEntityData.SitesData
+import org.wordpress.android.provider.query.QueryResult
+import org.wordpress.android.util.publicdata.WordPressPublicData
+import org.wordpress.android.viewmodel.ContextProvider
+import javax.inject.Inject
+
+private fun ContentResolver.query(
+    builder: Uri.Builder,
+    entityType: LocalContentEntity,
+    siteId: Int?,
+    entityId: Int?,
+) : Cursor? {
+    val entityPath = entityType.getPathForContent(siteId, entityId)
+    builder.appendEncodedPath(entityPath)
+    return query(builder.build(), arrayOf(), "", arrayOf(), "")
+}
+
+
+class LocalMigrationContentResolver @Inject constructor(
+    private val contextProvider: ContextProvider,
+    private val wordPressPublicData: WordPressPublicData,
+    private val queryResult: QueryResult,
+    private val dispatcher: Dispatcher,
+){
+    private inline fun <reified T : LocalContentEntityData> getDataForEntityType(
+        entityType: LocalContentEntity,
+        siteId: Int? = null,
+        entityId: Int? = null
+    ): T {
+        wordPressPublicData.currentPackageId().let { packageId ->
+            Uri.Builder().apply {
+                scheme("content")
+                authority("${packageId}.${LocalMigrationContentProvider::class.simpleName}")
+            }
+        }.let { uriBuilder ->
+            with (contextProvider.getContext().contentResolver) {
+                val cursor = query(uriBuilder, entityType, siteId, entityId)!!
```

Comment:
Would it make sense to replace the non-null assertion operator with something else to avoid throwing an NPE in case `query` returns `null`?

Reply:
Indeed. It makes sense to consider why this would be null and define error types for these cases. Same for the other `!!`. The nice thing about `!!` is that it's easy to spot in this WIP.. hence why I used it everywhere :sweat_smile: . Ultimately, we'll want to handle these gracefully - perhaps with an error class?

Reply:
> Ultimately, we'll want to handle these gracefully - perhaps with an error class?
    
  Sounds like a good idea 👍 🎉

Reply:
I've updated these double-bangs with better error messaging here: https://github.com/wordpress-mobile/WordPress-Android/pull/17398/commits/0f1b8de2f4072d8fdf489913c425268e9dfbacaf. I was thinking we can tackle a more unified error class in a later PR. Wdyt?